src/pages/gen2/build-a-backend/data/custom-business-logic/index.mdx

```diff
@@ -169,72 +167,192 @@ export const data = defineData({
     defaultAuthorizationMode: 'apiKey',
     apiKeyAuthorizationMode: {
       expiresInDays: 30
-    }
+    },
   },
   functions: {
-    // Step 3 - Map the function key to the function definition
-    echoHandler: defineFunction({
-      entry: './echo-handler.ts'
-    })
+    'echoHandler': echoHandler
   }
 });
 ```
 
-## Step 3 - Invoke the custom query or mutation
+</Block>
+<Block name="Custom resolver powered by AppSync JavaScript resolvers">
 
-<Callout info>
+Custom resolvers work on a "request/response" basis, you choose a data source, map your request to the data sources' input parameters, and then map the data source's response back to the query/mutation's return type. Custom resolvers provide the benefit of no cold starts, less infrastructure to manage, and no additional charge for Lambda function's execution. Review [Choosing between custom resolver and function](https://docs.aws.amazon.com/appsync/latest/devguide/resolver-reference-overview-js.html#choosing-data-source).
 
-**Coming soon:** The following example requires you to manually generate GraphQL client code as an escape hatch to invoke the custom query or mutation. Amplify Data is built with GraphQL under the hood. While you're not expected to know GraphQL or have to interface with GraphQL directly when Amplify (Gen 2) becomes generally available, it is helpful to understand the [underlying GraphQL concepts](https://graphql.org) that enable Amplify Data's functionality.
+In your **amplify/data/resource.ts** file, define a custom handler using `a.handler.custom`.
+
+```ts title="amplify/data/resource.ts"
+import {
+  type ClientSchema,
+  a,
+  defineData,
+} from '@aws-amplify/backend';
 
-<br />
-For Gen 2's general availability, we plan on integrating this first class into the
-Data client, so you can call a custom query or mutation with code, such as:
+const schema = a.schema({
+  Post: a.model({
+    content: a.string(),
+    likes: a.integer()
+      .authorization([a.allow.private().to(['read'])])
+  }).authorization([
+    a.allow.owner(),
+    a.allow.private().to(['read'])
+  ]),
 
-```ts
-const response = await client.queries.echo({ content: 'hello' });
+  likePost: a
+    .mutation()
+    .arguments({ postId: a.id() })
+    .returns(a.ref('Post'))
+    .authorization([a.allow.private()])
+    .handler(a.handler.custom({
+      dataSource: a.ref('Post'),
+      entry: './increment-like.js'
+    }))
+});
+
+export type Schema = ClientSchema<typeof schema>;
+
+export const data = defineData({
+  schema,
+  authorizationModes: {
+    defaultAuthorizationMode: 'apiKey',
+    apiKeyAuthorizationMode: {
+      expiresInDays: 30
+    }
+  },
+});
 ```
 
-</Callout>
+```ts title="amplify/data/increment-like.js"
```

Comment:
```suggestion
```ts title="amplify/data/increment-like.ts"
```

typescript to account for the Context type, should this also include the `Context` import?

Reply:
it needs to be JS resolver for now + using JSDocs.